src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
 /**
  * Parameterize this service and add its implementation in a separate class.
  */
-public interface MachineService {
+public interface MachineService<T> {
```

Comment:
When you parameterize interface MachineService keep in mind that we want to restrict types that can be used with it.

Not allow:
`MachineServiceImpl implements MachineService<Dog>`
Allow:
`MachineServiceImpl implements MachineService<Truck>`

```suggestion
public interface MachineService<T extends Machine> {
```

Reply:
++